internal/proxy/proxy.go

```diff
@@ -192,165 +192,189 @@ func (p *Proxy) Handle(w http.ResponseWriter, r *http.Request) {
 	}).ServeHTTP(w, r)
 }
 
-func (p *Proxy) handleJenkinsUIRequest(w http.ResponseWriter, r *http.Request, requestLogEntry *log.Entry) (cacheKey string, ns string, noProxy bool) {
+func (p *Proxy) handleJenkinsUIRequest(responseWriter http.ResponseWriter, request *http.Request, requestLogEntry *log.Entry) (cacheKey string, namespace string, noProxy bool) {
 	//redirect determines if we need to redirect to auth service
 	needsAuth := true
 	noProxy = true
 
-	//redirectURL is used for auth service as a target of successful auth redirect
-	redirectURL, err := url.ParseRequestURI(fmt.Sprintf("%s%s", strings.TrimRight(p.redirect, "/"), r.URL.Path))
+	// redirectURL is used for auth service as a target of successful auth redirect
+	redirectURL, err := url.ParseRequestURI(fmt.Sprintf("%s%s", strings.TrimRight(p.redirect, "/"), request.URL.Path))
 	if err != nil {
-		p.HandleError(w, err, requestLogEntry)
+		p.HandleError(responseWriter, err, requestLogEntry)
 		return
 	}
 
 	//If the user provides OSO token, we can directly proxy
-	if _, ok := r.Header["Authorization"]; ok { //FIXME Do we need this?
+	if _, ok := request.Header["Authorization"]; ok { //FIXME Do we need this?
```

Comment:
For the record, this can be removed. Needs to be tested though.

Reply:
https://github.com/fabric8-services/fabric8-jenkins-proxy/pull/198/commits/b4bc94048cc26b388ce3358165760aff2459154f